components/eam/src/chemistry/utils/read_spa_data.F90

```diff
@@ -0,0 +1,93 @@
+module read_spa_data
+
+!BSINGH - I didn't add any comments here due to time constrain but I will add comments later
+
+use tracer_data
+
+implicit none
+
+private
+
+public :: read_spa_data_init, read_spa_data_adv, read_spa_data_register
+
+type(trfld), pointer :: spa_fields(:)
+type(trfile)         :: spa_file
+logical              :: rmv_file = .false.
```

Comment:
It would be good to include a comment here saying what rmv_file means. I suspect it means "don't delete the file after pulling its data"?